// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option objc_class_prefix = "EDT";

package editor;

// The editing service definition.
service Editor {
    // Sends a command
    rpc SendCommand (Command) returns (CommandStatus) {}
}

enum Operation {
    INS = 0;
    DEL = 1;
}

message Command {
    Operation type = 1;
    uint32 position = 2;
    uint32 timeStamp = 3;
    uint32 userID = 4;
}

// The response message containing the operation status
message CommandStatus {
    uint32 status = 1;
}

